src/OmnichannelChatSDK.ts

```diff
@@ -1693,6 +1635,146 @@ class OmnichannelChatSDK {
         }
     }
 
+    public async getAgentAvailability(optionalParams: GetAgantAvailabilityOptionalParams = {}): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
+        const requestId = uuidv4();
+        
+        this.scenarioMarker.startScenario(TelemetryEvent.GetAgentAvailability, {
+            RequestId: requestId
+        });
+
+        if (!this.authSettings) {
+            const exceptionDetails: ChatSDKExceptionDetails = {
+                response: "UnAuthUnsupported",
+                message: "GetAgentAvailability is supported only for authenticated live chat widget."
+            };
+
+            this.scenarioMarker.failScenario(TelemetryEvent.GetAgentAvailability, {
+                RequestId: requestId,
```

Comment:
Can we add Chat Id here ?

Reply:
This API is called before startChat, so no Chat Id is available yet